app/uk/gov/hmrc/exports/repositories/SubmissionRepository.scala

```diff
@@ -59,6 +62,13 @@ class SubmissionRepository @Inject()(implicit mc: ReactiveMongoComponent, ec: Ex
       .collect(maxDocs = -1, FailOnError[Seq[Submission]]())
   }
 
+  def findOrCreate(eori: Eori, id: String, onMissing: Submission): Future[Submission] = {
```

Comment:
Sounds more like a service method to me rather than a repository one

Reply:
But logic is on repo level it self just combination of find and insert.